application/backend/src/services/training_service.py

```diff
@@ -105,6 +106,7 @@ async def _run_training_job(cls, job: Job, job_service: JobService) -> Model | N
                     model=model,
                     device=device,
                     synchronization_parameters=synchronization_parameters,
+                    max_epochs=max_epochs,
```

Comment:
Should we add this to the model training dialog? cc @ashwinvaidya17, @MarkRedeman

Reply:
It would be nice if we can already expose this (perhaps next to the picker of the training device from the new designs)

However there are some edge cases we would need to deal with. Some models will overwrite the max epochs to be 1 since they only need to do a single pass.
We could hardcode the UI to not show the input for those models, but that might get messy.

Reply:
Let's expose this and just have a note saying something like, "some models only extract features so max_epochs will be overridden for those models".